accessible/toolbox_treeview.component.js

```diff
@@ -0,0 +1,188 @@
+              </li>
+            </ol>
+            {{addClass(listItem, "blocklyHasChildren")}}
+          </li>
+          <div *ngFor="#inputBlock of block.inputList; #i=index">
+            <field-view [attr.aria-level]="level+1" *ngFor="#field of getInfo(inputBlock); #j=index" [field]="field" [level]="level+1"></field-view>
+            <toolbox-tree-view *ngIf="inputBlock.connection && inputBlock.connection.targetBlock()" [block]="inputBlock.connection.targetBlock()" [displayBlockMenu]="false" [level]="level+1"></toolbox-tree-view>
+            <li aria-selected=false #listItem1 role="treeitem" [attr.aria-level]="level+1" id="{{treeService.createId(listItem1)}}" *ngIf="inputBlock.connection && !inputBlock.connection.targetBlock()">
+              <label #label id="{{treeService.createId(label)}}">{{getInputTypeLabel(inputBlock.connection)}} {{valueOrStatement(inputBlock)}} needed:</label>
+              {{setLabelledBy(listItem1, concatStringWithSpaces("blockly-argument-text", label.id))}}
+            </li>
+          </div>
+        </ol>
+      </li>
+      <toolbox-tree-view *ngIf= "block.nextConnection && block.nextConnection.targetBlock()" [level]="level" [block]="block.nextConnection.targetBlock()" [displayBlockMenu]="false"></toolbox-tree-view>
+    `,
+    directives: [ng.core.forwardRef(
+        function() { return blocklyApp.ToolboxTreeView; }),
+        blocklyApp.FieldView],
+    inputs: ['block', 'displayBlockMenu', 'level', 'index', 'tree'],
+  })
+  .Class({
+    constructor: [blocklyApp.ClipboardService, blocklyApp.TreeService,
+                  function(_clipboardService, _treeService) {
+      this.infoBlocks = {};
+      this.nextBlock = {};
+      this.sharedClipboardService = _clipboardService;
+      this.treeService = _treeService;
+    }],
+    setLabelledBy: function(item, id) {
+      if (!item.getAttribute('aria-labelledby')) {
+        item.setAttribute('aria-labelledby', id);
+      }
+    },
+    concatStringWithSpaces: function() {
+      var string = arguments[0];
+      for (i = 1; i < arguments.length; i++) {
+        var arg = arguments[i];
+        // arg can be undefined and if so, should be ignored.
+        if (arg) {
+          string += ' ' + arg;
+        }
+      }
+      return string;
+    },
+    createCategoryDependantId: function(index, parentList) {
+      // If this is the first block in a category-less toolbox,
+      // the id should be blockly-toolbox-tree-node0.
+      if (index === 0) {
+        if (this.tree.getAttribute('aria-activedescendant') ==
+            'blockly-toolbox-tree-node0') {
+          this.addClass(parentList, 'blocklyActiveDescendant');
+          parentList.setAttribute('aria-selected', 'true');
+          this.treeService.setActiveDesc(parentList, this.tree.id);
+        }
+        return 'blockly-toolbox-tree-node0';
+      } else {
+        return this.treeService.createId(parentList);
+      }
+    },
+    addClass: function(node, classText) {
```

Comment:
See ngClass if you want to apply conditional classes. These are added in addition to the classes hardcoded in the class attribute.

Having said that, I don't entirely understand why some of the calls to this function can't be handled directly. E.g. you have `{{addClass(listItem, "blocklyHasChildren")}}` in the template. Why can't blocklyHasChildren just be written directly into the li?